### PR TITLE
Allow querying for an evaluation's information

### PR DESCRIPTION
**Pull request by lowjoel**
Also implement pulling the package.

### CODE REVIEW

File: app/controllers/course/assessment/programming_evaluations_controller.rb [hunk 1]
```diff
@@ -8,21 +8,29 @@
...
   end
 
   def allocate
-    fail ActiveRecord::RecordNotFound unless @programming_evaluation
-    @programming_evaluation.assign!(current_user)
-    response.status = :bad_request unless @programming_evaluation.save
+    save_success = @programming_evaluations.
+                   each { |evaluation| evaluation.assign!(current_user) }.
+                   map(&:save)
+    response.status = :bad_request unless save_success.reduce(true, &:&)
```
Comment (allenwq) [inline app/controllers/course/assessment/programming_evaluations_controller.rb]: Think `save_success.all?` would be more clear ?

Comment (lowjoel) [inline app/controllers/course/assessment/programming_evaluations_controller.rb]: darn I forgot about this in the docs:
> Passes each element of the collection to the given block. The method returns true if the block never returns false or nil. If the block is not given, Ruby adds an implicit block of { |obj| obj } which will cause all? to return true when none of the collection members are false or nil.

Comment (lowjoel) [inline app/controllers/course/assessment/programming_evaluations_controller.rb]: thanks!

**Comment by lowjoel** [comment]
@allenwq got it!

**Comment by allenwq** [comment]
@lowjoel Thanks !

**Pull request merged**